packages/coverage-v8/src/provider.ts

```diff
@@ -497,27 +486,49 @@ export class V8CoverageProvider extends BaseCoverageProvider implements Coverage
 
   private async convertCoverage(
     coverage: RawCoverage,
-    projectName?: ProjectName,
-    transformMode?: 'web' | 'ssr',
+    project: WorkspaceProject = this.ctx.getCoreWorkspaceProject(),
+    transformMode?: keyof CoverageFilesByTransformMode,
   ): Promise<CoverageMap> {
-    const viteNode
-      = this.ctx.projects.find(project => project.getName() === projectName)
-        ?.vitenode || this.ctx.vitenode
-    const fetchCache = transformMode
-      ? viteNode.fetchCaches[transformMode]
-      : viteNode.fetchCache
+    let fetchCache = project.vitenode.fetchCache
+
+    if (transformMode) {
+      fetchCache = transformMode === 'browser' ? new Map() : project.vitenode.fetchCaches[transformMode]
+    }
+
     const transformResults = normalizeTransformResults(fetchCache)
 
-    const scriptCoverages = coverage.result.filter(result =>
-      this.testExclude.shouldInstrument(fileURLToPath(result.url)),
-    )
+    async function onTransform(filepath: string) {
+      if (transformMode === 'browser' && project.browser) {
+        const result = await project.browser.vite.transformRequest(filepath.replace(project.config.root, ''))
+
+        if (result) {
+          return { ...result, code: `${result.code}// <inline-source-map>` }
+        }
+      }
+      return project.vitenode.transformRequest(filepath)
+    }
+
+    const scriptCoverages = []
+
+    for (const result of coverage.result) {
+      if (transformMode === 'browser') {
+        if (result.url.includes('@fs/')) {
+          result.url = `file://${result.url.replace('@fs/', '')}`
```

Comment:
Can we use `pathToFileURL` instead? In that case, we also need to replace `/@fs/` with `isWindows ? '' : '/'`

Reply:
Same reason as below - `pathToFleURL` escapes query parameters.